automation/tdx_canonical_verifier.sh

```diff
@@ -0,0 +1,184 @@
+ATTESTATION_DIR="$TDX_DIR"/attestation
+GUEST_TOOLS_DIR="$TDX_DIR"/guest-tools
+TD_GUEST_VERIFY_TEXT="tdx: Guest detected"
+TDT_HOST_VERIFY_TEXT="tdx: module initialized"
+SERVICE_ACTIVE_STATUS_VERIFY_TEXT="Active: active (running)"
+MPA_SERVICE_CHECK="mpa_registration_tool.service; enabled; preset: enabled"
+VERIFY_ATTESTATION_SCRIPT="verify_attestation_td_guest.sh"
+TD_GUEST_PASSWORD=123456
+TD_GUEST_PORT=10022
+MPA_REGISTRATION_CHECK="INFO: Registration Flow - Registration status indicates registration is completed successfully"
+TRUSTAUTHORITY_API_FILE="config.json"
+
+args=$(getopt -a -o i:h -l td_image_path: -- "$@")
+if [[ $? -gt 0 ]]; then
+  usage
+fi
+
+
+usage(){
+>&2 cat << EOF
+Usage: $0
+   [ -i | --td_image_path input ]
+EOF
+exit 1
+}
+
+
+clone_tdx_repo(){
+        echo -e "\n\nCloning TDX git repository with '$BRANCH_NAME' branch.."
+        [ -d $TDX_DIR ] && rm -rf $TDX_DIR
+        git clone -b $BRANCH_NAME https://github.com/canonical/tdx.git $TDX_DIR
+}
+
+
+verify_tdx_host(){
+        echo -e "\n\nVerifying whether host is configured with TDX or not.."
+        var="$(sudo dmesg | grep -i tdx)"
+        echo "$var"
+        if [[ "$var" =~ "${TDT_HOST_VERIFY_TEXT}" ]]; then
+                        echo -e "TDX is configured on the Host.\n"
+        else
+                        echo -e "Failure: TDX is not configured on the Host!!\n"
+                        exit 1
+        fi
+}
+
+verify_td_guest(){
+        echo -e "\n\nVerifying whether the guest is configured with TDX or not.."
+        echo "TD guest is running on portnumber : $TD_GUEST_PORT"
+        out=$(sshpass -p "$TD_GUEST_PASSWORD" ssh -o StrictHostKeyChecking=no -p $TD_GUEST_PORT root@localhost 'dmesg | grep -i tdx' 2>&1 )
+        echo "$out"
+        if [[ "$out" =~ "${TD_GUEST_VERIFY_TEXT}" ]]; then
+                        echo -e "TDX is configured on guest.\n"
+        else
+                        echo -e "Failure: TDX is not configured on guest!!\n"
+                        exit 1
+        fi
+}
+
+
+run_verify_td_guest(){
+        echo -e "\nCreating TD guest with QEMU\n"
+        sudo usermod -aG kvm $USER
+        cd "$GUEST_TOOLS_DIR"
+        var=$(TD_IMG=/home/sdp/bprashan/test/tdx-guest-ubuntu-24.04-generic.qcow2 ./run_td.sh)
+        ret=$?
+        echo -e "$var"
+        if [ $ret -ne 0 ]; then
+                        exit 1
+        fi
+        echo -e "\nVerifying TD guest on QEMU\n"
+        TD_GUEST_PORT=$(echo $var | awk -F '-p' '{print $2}' | cut -d ' ' -f 2)
+        verify_td_guest
+}
+
+check_production_system() {
+        cd $ATTESTATION_DIR
+        echo -e "\n\nVerifying whether the system supports attestation.."
+        output=$(sudo ./check-production.sh)
+        echo $output
+        if [[ $output =~ "Production" ]]; then
+                echo -e "Attestation is supported.\nThe attestation components are installed \n"
+        sed -i 's/^TDX_SETUP_ATTESTATION=0/TDX_SETUP_ATTESTATION=1/' "$TDX_DIR"/setup-tdx-config

```

Comment:
Do we need to modify the setup-tdx-config when the installation has already happened? I know this is a copy-paste from above but I feel this can be removed.

Reply:
added tdx-canonical-common file to maintain common functionality between setup and verifier script